### PR TITLE
Feat/auth

### PR DESCRIPTION
**Pull request by LeAlencar**

### CODE REVIEW

File: src/components/RegisterForm/index.tsx [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+              position: toast.POSITION.TOP_RIGHT,
+              autoClose: 2000,
+              hideProgressBar: false,
+              closeOnClick: true,
+              pauseOnHover: true,
+              draggable: true,
+              progress: undefined
+            })
+          }
+        }
```
Comment (fersilva16) [inline src/components/RegisterForm/index.tsx:57]: early return

File: src/context/AuthContext.tsx [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+
+export const AuthContext = createContext({} as IAuthContext)
+
+export const AuthProvider = ({ children }: { children: ReactNode }) => {
+  const token = GetToken()
+
+  const [isAuth, setIsAuth] = useState(token ? true : false)
+
+  useEffect(() => {
+    token ? setIsAuth(true) : localStorage.removeItem('login')
```
Comment (fersilva16) [inline src/context/AuthContext.tsx:18]: It should setIsAuth to false if doesn't have token too no?

File: src/context/AuthContext.tsx [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+interface IAuthContext {
+  isAuth: boolean
+  login: (token: string) => void
+  logout: () => void
+}
+
+export const AuthContext = createContext({} as IAuthContext)
+
+export const AuthProvider = ({ children }: { children: ReactNode }) => {
+  const token = GetToken()
```
Comment (fersilva16) [inline src/context/AuthContext.tsx:13]: You should validate if this token is valid in back end too

File: src/pages/Login.tsx [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+import { useNavigate } from 'react-router-dom'
+import { FormLogin } from '../components/LoginForm'
+import { AuthContext } from '../context/AuthContext'
+
+export const LoginPage = () => {
+  const { isAuth } = useContext(AuthContext)
+  const navigate = useNavigate()
+
+  useEffect(() => {
+    isAuth ? navigate('/') : ''
```
Comment (fersilva16) [inline src/pages/Login.tsx:11]: prefer `if (isAuth) navigate('/')`, it's easier to read

**Review by fersilva16 — COMMENTED** [review]
Prefer minor PRs: one subject, one commit, one PR

It makes the PR easier to review, more organized and atomic

File: src/components/RegisterForm/index.tsx (hunk 1 above)
Comment (LeAlencar) [reply src/components/RegisterForm/index.tsx:57]: what do you mean?

File: src/context/AuthContext.tsx (hunk 2 above)
Comment (LeAlencar) [reply src/context/AuthContext.tsx:18]: yeah, I forgot about it.

File: src/context/AuthContext.tsx (hunk 3 above)
Comment (LeAlencar) [reply src/context/AuthContext.tsx:13]: https://github.com/samueldurantes/xau/blob/master/packages/web/src/contexts/auth.tsx#L20 using something like that?

File: src/pages/Login.tsx (hunk 4 above)
Comment (LeAlencar) [reply src/pages/Login.tsx:11]: Ur right. Going to change it

**Comment by LeAlencar** [comment]
Thanks so much for the feedback

File: src/components/RegisterForm/index.tsx (hunk 1 above)
Comment (fersilva16) [reply src/components/RegisterForm/index.tsx:57]: Yep, in the backend

File: src/context/AuthContext.tsx (hunk 3 above)
Comment (fersilva16) [reply src/context/AuthContext.tsx:13]: Yep, in the backend

File: src/components/RegisterForm/index.tsx (hunk 1 above)
Comment (fersilva16) [reply src/components/RegisterForm/index.tsx:57]: if else is more complex to read than early return, prefer something like:
- you return earlier
- keep the simpler part in the if
- avoid unnecessary indentation and brackets

```ts
if (userCreate.error) {
  toast.error('This user already exists!', {
    position: toast.POSITION.TOP_RIGHT,
    autoClose: 2000,
    hideProgressBar: false,
    closeOnClick: true,
    pauseOnHover: true,
    draggable: true,
    progress: undefined
  })

  return;
}
  
const token = userCreate.token

login(token);

navigate('/');

toast.success('Logged in!', {
  position: toast.POSITION.TOP_RIGHT,
  autoClose: 2000,
  hideProgressBar: false,
  closeOnClick: true,
  pauseOnHover: true,
  draggable: true,
  progress: undefined
})
```

Comment (fersilva16) [reply src/components/RegisterForm/index.tsx:57]: In more complex cases it's way way better, like the one in this article: https://gomakethings.com/the-early-return-pattern-in-javascript

File: src/context/AuthContext.tsx (hunk 3 above)
Comment (LeAlencar) [reply src/context/AuthContext.tsx:13]: Provide me more context. When and where should I validate it? When I'm logging in I generate a new token and should validate it right away?

File: src/components/RegisterForm/index.tsx (hunk 1 above)
Comment (LeAlencar) [reply src/components/RegisterForm/index.tsx:57]: Hm, didn't know this was a pattern. Thanks for this, will read more about it.

**Pull request merged**